src/v/cloud_storage/tests/cloud_storage_e2e_test.cc

```diff
@@ -702,6 +702,14 @@ TEST_P(CloudStorageEndToEndManualTest, TestTimequeryAfterArchivalGC) {
       archiver->manifest().get_spillover_map().begin()->base_offset,
       first_seg.base_offset);
 
+    // Supply some phony disk stats so that the cache doesn't panic and
+    // think it has zero bytes of space. Otherwise we seem to be racing
+    // against an async notification in release mode.
+    app.shadow_index_cache.local().notify_disk_status(
+      100ULL * 1024 * 1024 * 1024,
+      50ULL * 1024 * 1024 * 1024,
+      storage::disk_space_alert::ok);
```

Comment:
I don't understand this at all, might be good to get someone with more context on this test to review :)

Reply:
could this have anything to do with the isolated test environment that bazel provides?

Reply:
maybe? but it's not deterministic and doesn't reproduce in debug mode afaict. still I don't _think_ this really affects the semantics of the test. wdyt?

Reply:
yeh, seems like a race

Reply:
@dotnwat you feel good to merge these or should i get a second opinion?

Reply:
It's odd that this isn't being seen at all in CMake. Agree with others this just seems like a race between local node monitor (on a 1s interval) and the test performing the trim. Wondering if maybe bazel's isolation means that the stat is slower or somesuch.

In any case, this fix lgtm; we do it in a few other tests